packages/server-wallet/src/models/objective.ts

```diff
@@ -27,8 +17,16 @@ function extractReferencedChannels(objective: ObjectiveType): string[] {
   }
 }
 
-export class ObjectiveChannel extends Model {
-  readonly objectiveId!: ObjectiveStoredInDB['objectiveId'];
+/*
+  A DBObjective is a wire objective with a status and an objectiveId
+*/
+export type DBObjective = Objective & {
+  objectiveId: string;
+  status: 'pending' | 'approved' | 'rejected' | 'failed' | 'succeeded';
+};
```

Comment:
I thought your aim was to just have `ObjectiveModel` (formerly `Objective) be this type?

Also: moving the type seems good. Renaming it seems unecessary?

Reply:
That was the goal. The problem with that is that if you have just the `ObjectiveModel`, you don't get any type narrowing, because you don't have a union type. So it seems like we need another thing.

I'm open to a different name. I thought that `ObjectiveStoredInDB` was a bit confusing, as that suggested that it should be what the `ObjectiveModel` actually is. That said I'm not sure that `DBObjective` is much better. Maybe it should actually be `Objective` and the wallet-core objective should become `WireObjective`?